src/datetime/serde.rs

```diff
@@ -1206,6 +1222,1021 @@ pub mod ts_seconds_option {
     }
 }
 
+/// Ser/de to/from timedeltas in nanoseconds
+///
+/// Intended for use with `serde`'s `with` attribute.
+///
+/// # Example:
+///
+/// ```rust
+/// # use chrono::TimeDelta;
+/// # use serde_derive::{Deserialize, Serialize};
+/// use chrono::serde::td_nanoseconds;
+/// #[derive(Deserialize, Serialize)]
+/// struct S {
+///     #[serde(with = "td_nanoseconds")]
+///     delta: TimeDelta,
+/// }
+///
+/// let time = TimeDelta::nanoseconds(2018517);
+/// let my_s = S { delta: delta.clone() };
+///
+/// let as_string = serde_json::to_string(&my_s)?;
+/// assert_eq!(as_string, r#"{"delta":2018517}"#);
+/// let my_s: S = serde_json::from_str(&as_string)?;
+/// assert_eq!(my_s.delta, delta);
+/// # Ok::<(), serde_json::Error>(())
+/// ```
+pub mod td_nanoseconds {
+    use core::fmt;
+    use serde::{de, ser};
+
+    use crate::serde::invalid_td;
+    use crate::TimeDelta;
+
+    use super::NanoSecondsTimedeltaVisitor;
+
+    /// Serialize a [`TimeDelta`] into an integer number of nanoseconds.
+    ///
+    /// Intended for use with `serde`s `serialize_with` attribute.
+    ///
+    /// # Errors
+    ///
+    /// This function returns an error on an out of range `TimeDelta`.
+    ///
+    /// # Example:
+    ///
+    /// ```rust
+    /// # use chrono::TimeDelta;
+    /// # use serde_derive::Serialize;
+    /// use chrono::serde::td_nanoseconds::serialize as to_nano_td;
+    /// #[derive(Serialize)]
+    /// struct S {
+    ///     #[serde(serialize_with = "to_nano_td")]
+    ///     delta: TimeDelta,
+    /// }
+    ///
+    /// let my_s = S {
+    ///     delta: TimeDelta::nanoseconds(2018517),
+    /// };
+    /// let as_string = serde_json::to_string(&my_s)?;
+    /// assert_eq!(as_string, r#"{"delta":2018517}"#);
+    /// # Ok::<(), serde_json::Error>(())
+    /// ```
+    pub fn serialize<S>(td: &TimeDelta, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: ser::Serializer,
+    {
+        serializer.serialize_i64(td.num_nanoseconds().ok_or(ser::Error::custom(
+            "value out of range for a timedelta with nanosecond precision",
```

Comment:
Let's capitalize `TimeDelta` in these error messages, too.

Reply:
Timestamp is one word in English, time delta is not, so there's a good reason to treat them differently. We should use `TimeDelta` consistently.